pkg/binding/binding.go

```diff
@@ -91,6 +91,10 @@ func (o *BindingClient) GetServiceInstances(namespace string) (map[string]unstru
 		// List all the instances of the restMapping object
 		resources, err := o.kubernetesClient.ListDynamicResources(namespace, restMapping.Resource)
 		if err != nil {
+			if kerrors.IsForbidden(err) {
```

Comment:
You probably also need to check if IsNotFound, as the namespace could be completely deleted

Reply:
Would it be possible to check this differently? Or maybe display the error to the user..

With this, if I run `odo add binding` with a user that does not have the required permissions to list the corresponding resources, I have this error:

```
❯ odo add binding     
? Do you want to list services from: current namespace
 ✗  No bindable service instances found in current namespace
```

On `main`, the error returned was much more explicit to me:
```
❯ odo add binding
? Do you want to list services from: current namespace
 ✗  perconaservermongodbs.psmdb.percona.com is forbidden: User "user" cannot list resource "perconaservermongodbs" in API group "psmdb.percona.com" in the namespace "user-ns"
```

Reply:
Or logging the error with `klog` could be sufficient to help with troubleshooting.

Reply:
`klog` sounds more appropriate to me. I have added it.